src/components/VideoCard/VideoCard.component.jsx

```diff
@@ -1,12 +1,34 @@
-import React from 'react';
-import { Container, Preview, Title, Body } from './VideoCard.styles';
-
-const VideoCard = ({ thumbnail, title, body }) => (
-  <Container>
-    <Preview src={thumbnail} />
-    <Title>{title}</Title>
-    <Body>{body}</Body>
-  </Container>
-);
+import React, { useContext } from 'react';
+import { Container, Thumbnail, Title, Description } from './VideoCard.styles';
+import AppContext from '../../providers/AppContext';
+
+const VideoCard = ({ videoId, title, description, thumbnail, onClickVideoHandler }) => {
+  const {
+    setSelectedVideoId,
+    setSelectedVideoTitle,
+    setSelectedVideoDescription,
+  } = useContext(AppContext);
+
+  // TODO: This is a great candidate to move to the context, but I need to figure out how to pass video ID because
```

Comment:
you could use the route param instead of passing it down as a prop, I'm not sure that it will prevent the cycling though